### PR TITLE
Proper identifier unquoting

### PR DESCRIPTION
**Pull request by sfc-gh-cgorrie**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added new automated tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [ ] I've described my changes in the release notes.
   * [x] I've described my changes in the section below.

### Changes description
- rename `identifier_as_part` to `unquote_identifier`
- use `unquote_identifier` for all LIKE queries + introspecting returned results so any identifier will work properly
- search in returned resultsets for the right row, rather than assuming it's the first one (due to LIKE being unable to restrict to exact case)
- fix tests so that unquoted identifiers are returned in UPPERCASE
- fix a bug in config merging uncovered by using quoted identifiers (`YAML` objects can be scalar)

### CODE REVIEW

File: src/snowcli/cli/nativeapp/manager.py [hunk 1]
```diff
@@ -319,23 +331,36 @@ def app_exists(self) -> bool:
...
         with self.use_role(self.app_role):
             show_app_cursor = self._execute_query(
-                f"show applications like '{self.app_name}'", cursor_class=DictCursor
+                f"show applications like '{unquote_identifier(self.app_name)}'",
+                cursor_class=DictCursor,
+            )
+            return (
+                find_row(
+                    show_app_cursor,
+                    lambda row: row[NAME_COL] == unquote_identifier(self.app_name),
```
Comment (sfc-gh-bgoel) [inline src/snowcli/cli/nativeapp/manager.py:340]: Should we make this into a utility function instead of defining it twice in lines 291 and 340?

**Review by sfc-gh-bgoel — APPROVED** [review]
LGTM, minor comment.

File: src/snowcli/cli/nativeapp/manager.py (hunk 1 above)
Comment (sfc-gh-cgorrie) [reply src/snowcli/cli/nativeapp/manager.py:340]: I debated it. One thing I didn't want to do was hide the fact that we were unquoting the identifier. We could do something like:
```python
find_row_matching(show_app_cursor, name=unquote_identifier(self.app_name))
```
But maybe that's too magical (unpacking kwargs as things that need to match in the dict)? What do you think?

Comment (sfc-gh-cgorrie) [reply src/snowcli/cli/nativeapp/manager.py:340]: I tried it out; it doesn't really help with readability and it makes it harder to use a constant for `NAME_COL`, so I think we should keep things as-is.

**Review by sfc-gh-turbaszek — APPROVED** [review]

**Pull request merged**